### ISSUE TITLE
This defaultMessage is extremely confusing

### ISSUE THREAD

**Issue by fujicoin** [trezor/trezor-suite#5087]
The defaultMessage for `TR_TOR_DISABLE_ONIONS_ONLY_NO_MORE_TITLE` is 'Custom Backends are no longer using onion addresses only.' now.
But do you want to say 'Use non-onion custom backends.' ? 

https://github.com/trezor/trezor-suite/blob/b6094d33db0cd2f9bba8fea300e14e6f81517e4a/packages/suite/src/support/messages.ts#L3205

**Comment by matejkriz** [comment]
Hi @fujicoin, thanks for the question. I believe, that comments directly in Crowdin is better place for this discussion, so I will close this issue and feel free to raise it there.

I agree, that the message is a bit confusing but the logic of it is accurate. We want to say, that you either add some non-onion custom backend or replaced the onion one with non-onion or switched to defaults...

**Comment by fujicoin** [comment]
@matejkriz 

This discussion is not possible in crowdin.

https://github.com/trezor/trezor-suite/blob/d623ba24777ee7dfa0e430f5151225f44b727d27/packages/suite/src/components/suite/modals/DisableTor.tsx#L115

The logic of L115-L158 in DisableTor.tsx is something unnatural.

The use of the variable name `onionBackends.length` seems inappropriate.
What number does `onionBackends.length` actually count?
As the name implies, if it's a number of onion backends, then all default messages in that range seem strange.

Since the defaultMessage for TR_TOR_DISABLE_ONIONS_ONLY is 'Missing non-onion Custom Backends', `onionBackends.length` seems to be related to the number of non-onion custom backends.
However, even assuming that, the logic of the ternary formula of L115-119 seems to be the opposite.

I have no choice but now I'm just looking at the defaultMessage flow to determine what the programmer wants to do.

The message for `TR_TOR_DISABLE_ONIONS_ONLY_NO_MORE_DESCRIPTION` on L145 is 'You can safely disable Tor now.'.
What does safety mean here?
I'm assuming it's connecting to a non-onion custom backend instead of the default Trezor backend.

Do not make a hasty decision and double-check the logic of L115-L158.

**Comment by fujicoin** [comment]
@matejkriz 

For your convenience, I replaced the id key in the range with the default message.

```
    return (
        <>
            <Modal
                cancelable
                onCancel={onCancel}
                heading={
                    onionBackends.length ? (
                        <Translation id="Missing non-onion Custom Backends" />
                    ) : (
                        <Translation id="Disable Tor" />
                    )
                }
            >
                {onionBackends.length ? (
                    <>
                        <Title>
                            <Translation id="Disabling Tor now will reset all Onion Backends to the default Trezor servers." />
                        </Title>
                        <Description>
                            <Translation id="Please add non-onion custom backend addresses to prevent this behavior." />
                        </Description>
                        {onionBackends.map(({ coin, urls }) => (
                            <BackendRow
                                key={coin}
                                coin={coin}
                                urls={urls}
                                onSettings={() => setCoin(coin)}
                            />
                        ))}
                    </>
                ) : (
                    <>
                        <Title>
                            <Translation id="Custom Backends are no longer using onion addresses only." />
                        </Title>
                        <Description>
                            <Translation id="You can safely disable Tor now." />
                        </Description>
                    </>
                )}
                <ButtonWrapper>
                    <Button
                        variant={onionBackends.length ? 'secondary' : 'primary'}
                        onClick={onDisableTor}
                    >
                        <Translation
                            id={
                                onionBackends.length
                                    ? 'Disable Tor and switch to Default Backends'
                                    : 'Disable Tor'
                            }
                        />
                    </Button>
                </ButtonWrapper>
            </Modal>
            {coin && <AdvancedCoinSettings coin={coin} onCancel={() => setCoin(undefined)} />}
        </>
    );
```

**Comment by matejkriz** [comment]
@fujicoin The naming is hard, so I can understand the confusion. But even after the double checking, the logic is correct now. 


`onionBackends.length` is number of coins with custom backends set up, that have only onion custom backends. 


For example:
enabled coins: BTC, LTC, ETH
BTC and ETH has custom backends, LTC on default backends:

BTC - only one onion custom backend http://btc1.trezoriovpjcahpzkrewelclulmszwbqpzmzgub37gbcjlvluxtruqad.onion
LTC - default backends
ETH- one onion custom backend http://eth1.trezoriovpjcahpzkrewelclulmszwbqpzmzgub37gbcjlvluxtruqad.onion and one non-onion https://eth1.trezor.io

so `onionBackends.length` is 1 as only BTC has onion-only custom backends...

`onionBackends`  looks like this then:
```json
[
    {
        "coin": "btc",
        "type": "blockbook",
        "urls": [
            "http://btc1.trezoriovpjcahpzkrewelclulmszwbqpzmzgub37gbcjlvluxtruqad.onion"
        ]
    }
]
```

**Comment by fujicoin** [comment]
Thank you for your explanation. I understand the logic.
I think you need a proper variable name in `onionBackends.length` or a comment that explains the meaning of the variable.

Given that the logic is correct, I think the English description of "Custom Backends are no longer using onion addresses only." is inappropriate.

I suggest the following description.
”The connection to onion custom backends will switch to non-onion custom backends.”

**Comment by fujicoin** [comment]
@matejkriz 

Thank you for your explanation. I understand the logic.
I think you need a proper variable name in `onionBackends.length` or a comment that explains the meaning of the variable.

Given that the logic is correct, I think the English description of "Custom Backends are no longer using onion addresses only." is inappropriate.
Perhaps users will not understand what this means.

I suggest the following description.
”The connection to onion custom backends will switch to non-onion custom backends.”

**Comment by matejkriz** [comment]
To be honest, this is an edge case of the edge case feature and I'm afraid we have a lot more important messages harder to understand :-( 
Your suggestion is easier to understand, but incorrect - at that point there are no more backends, that have only onion backends so nothing will switch. 
Have you tried going through it in the app?

**Comment by fujicoin** [comment]
@matejkriz 

I don't understand your explanation, "at that point there are no more backends, that have only onion backends so nothing will switch."

The absence of "coins that have only onion custom backends" means that all coins connected to the onion have non-onion custom backends.
Wouldn't it switch to those non-onion custom backends once Tor was turned off?

Here's a simpler way to write the logic:
```
onionBackends.length ?

case: coins that have only onion custom backends >= 1
"Missing non-onion Custom Backends"
"Disabling Tor now will reset all Onion Backends to the default Trezor servers."
"Please add non-onion custom backend addresses to prevent this behavior."
'Disable Tor and switch to Default Backends'

case: coins that have only onion custom backends = 0
"Disable Tor"
"Custom Backends are no longer using onion addresses only."	★
"You can safely disable Tor now."
'Disable Tor'
```

In the first place, the English sentence "Custom Backends are no longer using onion addresses only." is strange.
Why is "Custom Backends" the subject?
Is it possible to paraphrase this sentence with another subject?

**Comment by fujicoin** [comment]
I actually tested it as a custom backend using the link below.
```
https://btc2.trezor.io
http://btc4.trezoriovpjcahpzkrewelclulmszwbqpzmzgub37gbcjlvluxtruqad.onion
```

If Tor is turned off when the connection destination is onion link, the connection destination will switch to non-onion custom backend.
However, no message appeared no matter how many times I tried.

And if I set only onion custom backend and turn off Tor, the message "Missing non-onion Custom Backends" will appear.

This is what I claim to be.

**Comment by hynek-jina** [comment]
@fujicoin 
When you want to turn-off Tor, you have to specify non-onion address to remain on a custom backend. If you have a custom onion and non-onion address we simply switch you according to your Tor state (on/off)...
I don't see a problem here.

**Comment by fujicoin** [comment]
@matejkriz 

Certainly, if you set both custom onion and non-onion address, nothing will be displayed, so there is no actual problem.

Let's get away from the logic story.

Still, the English sentence "Custom Backends are no longer using onion addresses only." Is strange.
The subject of the present progressive tense "are using" should be "You" or "Suite" instead of "Custom Backends".

**Comment by jerechinsky** [comment]
I don't think the sentence is that weird. Perhaps the sentence could be rephrased a little like this: 

> Custom backends are no longer using onion addresses exclusively.

@b-irving, could you please chime in? 🙇‍♂️🙏 
Do you think my suggestion is good or do you have a better idea?

**Comment by fujicoin** [comment]
@matejkriz @alex-jerechinsky 

I researched various behaviors related to the Tor function and got a complete grasp.
I translated key "TR_TOR_DISABLE_ONIONS_ONLY_NO_MORE_TITLE" into Japanese as "オニオンカスタムバックエンドへの接続は、非オニオンカスタムバックエンドに切り替わります。".
The Japanese can fully understand the behavior that will occur in this translation.
Please understand that the Japanese cannot understand what will happen if the default English text is translated directly.
Let's end this discussion.
